examples/darkmode/src/components/ThemeManager.astro

```diff
@@ -0,0 +1,69 @@
+---
+type Props = {
+  "default-theme"?: "auto" | "dark" | "light";
+};
+
+const { "default-theme": defaultTheme = "auto" } = Astro.props;
+---
+
+<theme-manager data-default-theme={defaultTheme}></theme-manager>
+<script is:inline>
+  if (!customElements.get("theme-manager")) {
+    class ThemeManager extends HTMLElement {
+      connectedCallback() {
+        this.initThemeManagement(this.dataset.defaultTheme);
```

Comment:
Is there a risk of making `window.theme` not available before other js using it? I guess not since it's not a bundled script but I'm not familiar with web components

Reply:
Yes that could be a problem for some, that's why it goes in the `<head>` of layout but even still it probably could be emphasized in the tip better